tools/vm-migrator/migrator/subnet_region_migrator.py

```diff
@@ -0,0 +1,502 @@
+
+    with open(source_file, 'r') as csvfile:
+        csv_dict_reader = DictReader(csvfile)
+        for row in csv_dict_reader:
+            if (row['name'] in machine_names_to_filter):
+                filtered.append(row)
+
+    overrtie_file = query_yes_no(
+        "About to overrite %s with %i records, please confirm to continue" %
+        (destination_file, len(filtered)),
+        default='no')
+
+    if (overrtie_file):
+        with open(destination_file, 'w') as csvfile:
+            writer = DictWriter(csvfile, fieldnames=headers)
+            writer.writeheader()
+            writer.writerows(filtered)
+    return overrtie_file
+
+
+def release_ips_from_file(file_name):
+    with open(file_name, 'r') as read_obj:
+        csv_dict_reader = DictReader(read_obj)
+        for row in csv_dict_reader:
+            parsed_link = instance.parse_self_link(row['self_link'])
+            region = instance.get_region_from_zone(parsed_link['zone'])
+            project = parsed_link['project']
+            ips = []
+            # The first ip for a machine is reserved with the same name as the VM
+            ips.append(row['name'])
+            # Find the reserved alias ips
+            for i in range(4):
+                ip_name = row.get('alias_ip_name_' + str(i + 1))
+                if (ip_name != '' and row['range_name_' + str(i + 1)] == ""):
+                    ips.append(ip_name)
+            subnet.release_specific_ips(project, region, ips)
+            time.sleep(2) # Prevent making too many requests in loop
+
+
+# main function
+def main(project,
+         source_zone,
+         source_zone_2,
+         source_zone_3,
+         source_subnet,
+         machine_image_region,
+         target_subnet,
+         target_region,
+         source_region,
+         subnet_name,
+         step,
+         log,
+         file_name='export.csv',
+         filter_file_name='filter.csv'):
+    """
+    The main method to trigger the VM migration
+    """
+    numeric_level = getattr(logging, log.upper(), None)
+    if not isinstance(numeric_level, int):
+        raise ValueError('Invalid log level: %s' % log)
+    logging.basicConfig(filename='migrator.log',
+                        format='%(asctime)s  %(levelname)s %(message)s',
+                        level=numeric_level)
+
+    logging.info("executing step %s" % (step))
+    if (step == 'prepare_inventory'):
+        logging.info("Preparing the inventory to be exported")
+        subnet.export_instances(project, source_zone, source_zone_2,
+                                source_zone_3, source_subnet, "source.csv")
+    if (step == 'filter_inventory'):
+        logging.info("Preparing the inventory to be exported")
+        subnet.export_instances(project, source_zone, source_zone_2,
+                                source_zone_3, source_subnet, "source.csv")
+        logging.info("filtering out the inventory")
+        overrite_file = filter_records("source.csv", filter_file_name,
```

Comment:
overwrite_file? (possible misspelling of variable name)

Reply:
fixed